src/tox/config/__init__.py

```diff
@@ -1056,7 +1061,7 @@ class ParseIni(object):
     def __init__(self, config, ini_path, ini_data):  # noqa
         config.toxinipath = ini_path
         using("tox.ini: {} (pid {})".format(config.toxinipath, os.getpid()))
-        config.toxinidir = config.toxinipath.dirpath()
+        config.toxinidir = config.toxinipath.dirpath() if ini_path.check(file=True) else ini_path
```

Comment:
Note that this feels a bit like a hack. I've tried to pass `None` and adapt the code to not do certain things if this is `None`, but it was too messy.